### PR TITLE
Distributed tracing and Red Hat build of OpenTelemetry 3.1 release

### PR DESCRIPTION
**Pull request by max-cx**
:warning: We are aware of the merge freeze in effect. However, this PR is tied to a product release date, which is currently set to February 26.

:bulb: Please note that the `XXL` size of this PR is misleading: content from many pages was collected into fewer pages (the pages with the release notes for past releases), and several sections were migrated into separate pages (OpenTelemetry). So this shows up like a lot of new content in the diffs, which is in fact not the case.

:warning: This PR aggregates cherry-picked commits from the following peer-reviewed PRs:

https://github.com/openshift/openshift-docs/pull/70943
https://github.com/openshift/openshift-docs/pull/71565
https://github.com/openshift/openshift-docs/pull/69962
https://github.com/openshift/openshift-docs/pull/71074

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16

Manual cherry-picks to OCP 4.12, 4.13, 4.14:
https://github.com/openshift/openshift-docs/pull/72123/
https://github.com/openshift/openshift-docs/pull/72127/
https://github.com/openshift/openshift-docs/pull/72128/

Issue:

https://issues.redhat.com/browse/OBSDOCS-764
https://issues.redhat.com/browse/OBSDOCS-790
https://issues.redhat.com/browse/OBSDOCS-811
https://issues.redhat.com/browse/TRACING-3574
https://issues.redhat.com/browse/TRACING-3802

Link to docs preview:

New content:
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/distr_tracing/distr_tracing_rn/distr-tracing-rn-3-1
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel_rn/otel-rn-3.1
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/distr_tracing/distr_tracing_tempo/distr-tracing-tempo-configuring#span-red-metrics-and-alerting-rules
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel-configuration-of-otel-collector#otel-config-target-allocator_otel-configuration-of-otel-collector

Reorganized content:
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/distr_tracing/distr_tracing_rn/distr-tracing-rn-past-releases
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel_rn/otel-rn-past-releases
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel-configuration-of-otel-collector
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel-configuration-of-instrumentation
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel-sending-traces-and-metrics-to-otel-collector
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel-config-send-metrics-monitoring-stack
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel-forwarding
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel-configuring-otelcol-metrics
https://72079--ocpdocs-pr.netlify.app/openshift-enterprise/latest/otel/otel-config-multicluster

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by max-cx** [comment]
/label peer-review-done
(see description for an explanation)

**Comment by max-cx** [comment]
@kalexand-rh, yes, I aggregated the PRs because
1. All of this content has to be merged as one single 3.1 docs release (imagine merging bits of the OCP 4.15 release in different PRs at different times).
2. This one PR avoids merge conflicts among the incorporated PRs.
3. The commits of the incorporated PRs have been cherry-picked in this branch, so the changes are the same.
4. The amount of changes is the same as is the amount of effort to review them, whether it's several PRs or one PR.

File: modules/distr-tracing-architecture.adoc [hunk 1]
```diff
@@ -43,3 +27,18 @@ This module included in the following assemblies:
...
+
+** *Agent* (Jaeger agent, Server Queue, Processor Workers) - The {JaegerShortName} agent is a network daemon that listens for spans sent over User Datagram Protocol (UDP), which it batches and sends to the Collector. The agent is meant to be placed on the same host as the instrumented application. This is typically accomplished by having a sidecar in container environments such as Kubernetes.
+
+** *Jaeger Collector* (Collector, Queue, Workers) - Similar to the Jaeger agent, the Jaeger Collector receives spans and places them in an internal queue for processing. This allows the Jaeger Collector to return immediately to the client/agent instead of waiting for the span to make its way to the storage.
+
+** *Storage* (Data Store) - Collectors require a persistent storage backend. {JaegerName} has a pluggable mechanism for span storage. Note that for this release, the only supported storage is Elasticsearch.
```
Comment (kcarmichael08) [inline modules/distr-tracing-architecture.adoc:38]: (Note that for this release): Is this release-specific documentation? If not I would consider removing that part and just going with "The only supported storage is ..."

File: modules/distr-tracing-tempo-config-spanmetrics.adoc [hunk 2]
```diff
@@ -92,3 +94,55 @@ spec:
 ----
 <1> Enables the monitoring tab in the Jaeger console.
 <2> The service name for Thanos Querier from user-workload monitoring.
+
+== Span RED metrics and alerting rules
```
Comment (kcarmichael08) [inline modules/distr-tracing-tempo-config-spanmetrics.adoc:98]: Needs an anchor ID

File: modules/otel-config-instrumentation.adoc [hunk 3]
```diff
@@ -97,10 +98,12 @@ spec:
...
 
+[id="otel-using-instrumentation-cr-with-service-mesh_{context}"]
 == Using the instrumentation CR with Service Mesh
 
 When using the instrumentation custom resource (CR) with {SMProductName}, you must use the `b3multi` propagator.
 
+[id="otel-configuration-of-apache-http-server-auto-instrumentation_{context}"]
 === Configuration of the Apache HTTP Server auto-instrumentation
 
 .Prameters for the `+.spec.apacheHttpd+` field
```
Comment (kcarmichael08) [inline modules/otel-config-instrumentation.adoc:109]:
```suggestion
.Parameters for the `+.spec.apacheHttpd+` field
```

**Review by kcarmichael08 — COMMENTED** [review]
First batch of comments

File: distr_tracing/distr_tracing_arch/distr-tracing-architecture.adoc [hunk 4]
```diff
@@ -27,4 +27,5 @@ include::modules/distr-tracing-architecture.adoc[leveloffset=+1]
 [id="additional-resources_distributed-tracing-architecture"]
 == Additional resources
 
-* xref:../../otel/otel-release-notes.adoc[{OTELName}]
+// xreffing to the installation page until further notice because OTEL content is currently planned for internal restructuring across pages that is likely to result in renamed page files
+* xref:../../otel/otel-installing.adoc[{OTELName}]
```
Comment (kcarmichael08) [inline distr_tracing/distr_tracing_arch/distr-tracing-architecture.adoc:31]: This needs an anchor ID in the xref per [guidelines]([distr_tracing/distr_tracing_arch/distr-tracing-architecture.adoc](https://github.com/openshift/openshift-docs/pull/72079/files#diff-6ec699242de42a69712e13ee6cf9f4a288f58b7d1acedad41cbc30b89063db30)).

File: modules/distr-tracing-architecture.adoc (hunk 1 above)
Comment (max-cx) [reply modules/distr-tracing-architecture.adoc:38]: Thank you. Fixing.

File: modules/distr-tracing-tempo-config-spanmetrics.adoc (hunk 2 above)
Comment (max-cx) [reply modules/distr-tracing-tempo-config-spanmetrics.adoc:98]: Thank you. Fixing.

**Review by kcarmichael08 — COMMENTED** [review]
Can you go through and make sure that all of your xrefs contain Anchor IDs? They will work on docs.openshift.com usually without this, but I think they can break the customer portal or not work in certain situations if they don't have this.

File: otel/otel-config-multicluster.adoc [hunk 5]
```diff
@@ -1,10 +1,8 @@
-// Module included in the following assemblies:
-// 
-// * otel/otel-configuring.adoc
-
 :_mod-docs-content-type: PROCEDURE
```
Comment (kcarmichael08) [inline otel/otel-config-multicluster.adoc:1]:
```suggestion
:_mod-docs-content-type: ASSEMBLY
```

File: modules/otel-config-instrumentation.adoc (hunk 3 above)
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:109]: Thank you. Fixing it.

File: otel/otel-configuring-otelcol-metrics.adoc [hunk 6]
```diff
@@ -1,10 +1,12 @@
-// Module included in the following assemblies:
-//
-// * otel/otel-configuring.adoc
-
 :_mod-docs-content-type: PROCEDURE
```
Comment (kcarmichael08) [inline otel/otel-configuring-otelcol-metrics.adoc:1]:
```suggestion
:_mod-docs-content-type: ASSEMBLY
```

File: otel/otel-config-multicluster.adoc (hunk 5 above)
Comment (max-cx) [reply otel/otel-config-multicluster.adoc:1]: OK. Done.

**Review by kcarmichael08 — COMMENTED** [review]
next batch of comments (above)

File: welcome/index.adoc
Comment (kcarmichael08) [inline welcome/index.adoc:1]: these also need anchor IDs for the xrefs

File: observability/index.adoc [hunk 7]
```diff
@@ -45,8 +45,8 @@ For more information, see xref:../distr_tracing/distr_tracing_arch/distr-tracing
 == {OTELName}
 Instrument, generate, collect, and export telemetry traces, metrics, and logs to analyze and understand your software's performance and behavior. Use open source backends like Tempo or Prometheus, or use commercial offerings. Learn a single set of APIs and conventions, and own the data that you generate. 
 
-For more information, see xref:../otel/otel-release-notes.adoc#otel-release-notes[{OTELName} release notes].
-//after the file is added to the observability directory, update xref path to ../observability/otel/otel-release-notes.adoc#otel-release-notes[{OTELName} release notes]. 
+For more information, see xref:../otel/otel_rn/otel-rn-3.1.adoc[{OTELName}].
```
Comment (kcarmichael08) [inline observability/index.adoc:48]: need anchor ID for xref

File: otel/otel-forwarding.adoc [hunk 8]
```diff
@@ -1,10 +1,8 @@
-// Module included in the following assemblies:
-//
-// * otel/otel-using.adoc
-
 :_mod-docs-content-type: PROCEDURE
```
Comment (kcarmichael08) [inline otel/otel-forwarding.adoc:1]:
```suggestion
:_mod-docs-content-type: ASSEMBLY
```

File: distr_tracing/distr_tracing_arch/distr-tracing-architecture.adoc (hunk 4 above)
Comment (max-cx) [reply distr_tracing/distr_tracing_arch/distr-tracing-architecture.adoc:31]: Thank you. I've fixed this.

**Review by kcarmichael08 — COMMENTED** [review]
I think you've already fixed some of these - if so, please ignore

File: observability/index.adoc (hunk 7 above)
Comment (max-cx) [reply observability/index.adoc:48]: Thank you. I've fixed this.

File: otel/otel-forwarding.adoc (hunk 8 above)
Comment (max-cx) [reply otel/otel-forwarding.adoc]: OK. Done.

**Review by kalexand-rh — APPROVED** [review]
When this passes merge review, it can be merged.

**Comment by kcarmichael08** [comment]
Good to merge when finishes building.

**Pull request merged**